java-checks-test-sources/src/main/files/non-compiling/checks/MapKeyNotComparableCheck.java

```diff
@@ -0,0 +1,41 @@
+package checks;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public class MapKeyNotComparableCheck {
+  abstract class B<T> {
+    Map<T, Object> nonComparable = new HashMap<>();
+  }
+
+  static class MyKeyType {
+    @Override
+    public String toString() {
+      return super.toString();
+    }
+  }
+
+  static class MyComparableType implements Comparable {
+    @Override
+    public int compareTo(@NotNull Object o) {
+      return 0;
+    }
+  }
+
+  static class Program {
+    Map<UnknownType, Object> nonComparable = new HashMap<>(); // Compliant false positive
```

Comment:
If it's compliant it's never a false positive, it could just be a false negative